harness/determined/pytorch/_pytorch_context.py

```diff
@@ -303,7 +317,9 @@ def set_profiler(self, *args: List[str], **kwargs: Any) -> None:
         when training.
         """
         self.profiler = torch.profiler.profile(
-            on_trace_ready=torch.profiler.tensorboard_trace_handler(str(get_base_path({}))),
+            on_trace_ready=torch.profiler.tensorboard_trace_handler(
+                str(tensorboard.get_base_path({}))
```

Comment:
`tensorboard.get_base_path({})` is an legacy call that shouldn't be made if we can at all help it.

Instead, can you use `self.get_tensorboard_path()`?

When `get_base_path()` was added, we didn't have this pattern of context objects for each Trial plugin, so the only way to give the user a path to write tensorboard files to was with a global function.  But that approach is pretty limited.  For example, `get_base_path()` currently reads only Horovod environment variables, not torch.distributed environment variables, or variables that may be set by any 3rd party distributed training launcher.